multiproject/extension.py

```diff
@@ -0,0 +1,104 @@
+from pathlib import Path
+
+from sphinx.config import CONFIG_FILENAME, eval_config_file
+from sphinx.util import logging
+
+from . import __version__, utils
+
+log = logging.getLogger(__name__)
+
+
+def _get_project_from_config(config):
+    project = utils.get_project(
+        config.multiproject_projects,
+        config.multiproject_env_var,
+    )
+    return project, config.multiproject_projects[project]
+
+
+def _override_srdir(app, config):
+    """
+    Override app.srcdir with the value from the current project.
+
+    If the project doesn't have a path,
+    the name of the project is used as the path.
+
+    Values can be absolute or relative to the original source dir.
+    """
+    project, options = _get_project_from_config(config)
+    log.info("Using project: %s", project)
+
+    original_srcdir = Path(app.srcdir)
+    new_srcdir = Path(options.get("path", project))
+    if not new_srcdir.is_absolute():
+        new_srcdir = original_srcdir / new_srcdir
+    app.srcdir = str(new_srcdir.resolve())
+
+
+def _override_config(app, config):
+    """Override the config with specific values from the current project."""
+    _, options = _get_project_from_config(config)
+    project_config = options.get("config", {})
+    if options.get("use_config_file", True):
+        config_file = Path(app.srcdir) / CONFIG_FILENAME
+        if config_file.exists():
+            project_config.update(
+                eval_config_file(
```

Comment:
:+1: I was originally describing directly doing an `import`, but this is a much better version of that.